steel/docs/steel-history.md

```diff
@@ -0,0 +1,94 @@
+
+## Overview
+
+The Steel history feature enables the developer to query view call state older than 24 hours while still using the same Steel commitment mechanism. This is done by separating the pinned block in Steel into two separate blocks: the execution block and the commitment block. 
+
+Without the Steel history feature enabled, the pinned block is *both* the block from which the view call state is retrieved and the block used for the Steel commitment:
+
+```rust
+// WITHOUT STEEL HISTORY
+// Create an EVM environment from that provider defaulting to the latest block.
+let mut env = EthEvmEnv::builder()
+    .provider(provider.clone())
+    .build()
+    .await?;
+
+// Preflight the call at the pinned block, default latest.
+let mut contract = Contract::preflight(args.token_contract, &mut env);
+```
+
+With the Steel history feature enabled, the developer must specify both the execution block *and* the commitment block:
+
+```rust
+/// WITH STEEL HISTORY
+let builder = EthEvmEnv::builder()
+    .provider(provider.clone())
+    .block_number_or_tag(args.execution_block);
+
+#[cfg(any(feature = "beacon", feature = "history"))]
+let builder = builder.beacon_api(args.beacon_api_url);
+#[cfg(feature = "history")]
+let builder = builder.commitment_block(args.commitment_block);
+
+let mut env = builder.build().await?;
+
+// Preflight the call at the execution block.
+let mut contract = Contract::preflight(args.token_contract, &mut env);
+```
+
+The execution block is the block from which the view call state is retrieved. The commitment block is the block used for the Steel commitment. The commitment block has to fall within the 24 hour time window necessary for Steel commitment validation on-chain, but *crucially* the execution block can go further back on the scale of days, weeks or even months. 
+
+## How does Steel history work?
+
+The execution and commitment blocks are fundamentally related; the execution block should always be an ancestor of the commitment block. Therefore, it is possible to prove that the commitment block block root is canonical with the execution block block root by validating each beacon block root in between the two blocks.
+
+Steel handles this verification automatically when the history feature is enabled. Steel history works backwards from the commitment block to the execution block with consecutive calls to the beacon roots contract; validating a beacon root is a single call for every 24 hours of history and this step takes approximately 1M cycles within the Steel guest. 
+
+Ultimately, Steel will check the integrity of the view call data in the execution block by:
+
+1. proving that the execution block is a canonical ancestor of the commitment block
+2. carrying out the standard Steel commitment validation on-chain proving the integrity of the block root for the commitment block
```

Comment:
Very strictly speaking, this step of verifying the standard Steel commitment is not part of Steel as it happens on-chain

Reply:
That's a good point, I have fixed this to reflect that change with:

> Ultimately, Steel will check the integrity of the view call data in the execution block by proving that the execution block is a canonical ancestor of the commitment block. 
Once on-chain, succesfully validating the Steel commitment will prove the integrity of the block root for the commitment block.